### PR TITLE
Typing indicator ellipses

### LINKED ISSUE #5623: [HOLD for payment 2021-11-11] Chat - "Typing" isn't visible when someone from another account has longer email is typing
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Lunch the app
2. Log in with any account ( Account A)
3. Log in with another account where you don't have any last and first name to sets up ( Account B)
4. Start typing from Account B to Account A
## Expected Result:
"Typing" is visible when someone from another account  has longer email without name is typing
## Actual Result:
"Typing" isn't visible when someone from another account has longer email without name is typing
## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS
- Android
- Mobile Web

**Version Number:** 1.1.4.0
**Reproducible in staging?:** Yes
**Reproducible in production?:**Yes
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
Bug5260547_Screenshot_20210930-192721_Gallery__1_

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

### PR DESCRIPTION
**Pull request by akshayasalvi**
### Details

- Added ellipses for long emails in typing indicator

### Fixed Issues

$ https://github.com/Expensify/App/issues/5623

### Tests

1. Tested email long email and display name
2. Tested for small email
3. Tested with multiple users

### QA Steps

1. Go to chat with any account A (or with a long email)
2. Go to chat with account B
3. Send chats from Account A to Account B, the typing indicator should show the trimmed username along with the `is typing` word visible
4. Test the same with small email, it shouldn't trim the email
5. For two or more users it should show `Multiple users typing...`

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

web-long-email-without-ellipses
web-long-email-with-ellipses
web-normal-email

#### Mobile Web

mweb-email-ellipses

#### Desktop

desktop-email

#### iOS

ios-email-ellipses

#### Android

android-email-ellipses

### CODE REVIEW

File: src/languages/en.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ export default {
         newMsg: ({count}) => `${count} new message${count > 1 ? 's' : ''}`,
     },
     reportTypingIndicator: {
-        isTyping: 'is typing...',
+        isTyping: 'is typing',
```
Comment (parasharrajat) [inline src/languages/en.js:163]: I think we should keep this as it is??

File: src/styles/styles.js [hunk 2]
```diff
@@ -2179,6 +2179,22 @@ const webViewStyles = {
...
+function getTypingIndicatorTextStyle(isSmallScreenWidth) {
+    return isSmallScreenWidth
+        ? {
+            maxWidth: 200,
+        }
+        : {
+            maxWidth: 500,
+        };
+}
+
```
Comment (parasharrajat) [inline src/styles/styles.js:2197]: Hmm. This is supposed to be fluid. Take max available space with hiding `is typing...`

File: src/pages/home/report/ReportTypingIndicator.js [hunk 3]
```diff
@@ -53,33 +55,25 @@ class ReportTypingIndicator extends React.Component {
                 return <View style={[styles.chatItemComposeSecondaryRow]} />;
             case 1:
                 return (
-                    <View style={[styles.chatItemComposeSecondaryRow]}>
+                    <View style={[styles.chatItemComposeSecondaryRow, styles.flexRow]}>
                         <Text
                             style={[
                                 styles.chatItemComposeSecondaryRowSubText,
                                 styles.chatItemComposeSecondaryRowOffset,
+                                getTypingIndicatorTextStyle(this.props.isSmallScreenWidth),
```
Comment (parasharrajat) [inline src/pages/home/report/ReportTypingIndicator.js:63]: We will not need it. See comment below...

File: src/pages/home/report/ReportTypingIndicator.js [hunk 4]
```diff
@@ -53,33 +55,25 @@ class ReportTypingIndicator extends React.Component {
...
-                    </View>
-                );
-            case 2:
-                return (
-                    <View style={[styles.chatItemComposeSecondaryRow]}>
                         <Text
                             style={[
                                 styles.chatItemComposeSecondaryRowSubText,
-                                styles.chatItemComposeSecondaryRowOffset,
+                                styles.ml1,
```
Comment (parasharrajat) [inline src/pages/home/report/ReportTypingIndicator.js:72]: I think we should take the natural gap of `1 char`.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/languages/en.js (hunk 1 above)
Comment (akshayasalvi) [reply src/languages/en.js]: I did add `...` back to `typing...`.

image

I am not sure if two `...` look fine. But upto you guys.

File: src/pages/home/report/ReportTypingIndicator.js [hunk 5]
```diff
@@ -108,6 +100,7 @@ ReportTypingIndicator.displayName = 'ReportTypingIndicator';
 
 export default compose(
     withLocalize,
+    withWindowDimensions,
```
Comment (parasharrajat) [inline src/pages/home/report/ReportTypingIndicator.js:103]: Seems like unsued now.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Review by parasharrajat — CHANGES_REQUESTED** [review]
@akshayasalvi Did you test it on all platforms after the recent changes? If so, please update the screenshots.

I see that It does not work well on Android.
Screenshot 2021-10-20 04:38:25
The name should not trim until you have consumed all the available space. In the screenshot, there is plenty of space after `is typing...` 
So the Name will be trimmed when `is Typing...` is close to the right edge where the composer ends.

**Review by NikkiWines — CHANGES_REQUESTED** [review]
Couple minor comments but it's looking great

**Review by parasharrajat — DISMISSED** [review]
LGTM. Tests well. Please refer to Nikki's comments and we are good here. Thanks for the changes.

**Review by NikkiWines — APPROVED** [review]
👍  Looks good!!

**Comment by NikkiWines** [comment]
@parasharrajat not sure if you wanted to give this another review or not. 

If you haven't by EOD, I'll merge 🚀

**Review by parasharrajat — APPROVED** [review]

**Comment by nazam100** [comment]
Accessibility Issue : Incorrect focus order.
WCAG Failure : 2.4.3.
Expected result : The focus should reach to the chat box after navigating from + button(new chat). .
Actual result :  The Focus is moved to menu 'Search tabs menu button subMenu'

Attachments : 

https://user-images.githubusercontent.com/66008453/140066656-57545f6b-8c04-4bc2-98da-874a66f05c3a.mp4

**Comment by nazam100** [comment]
Accessibility Issue : Incorrect Focus order through messages is reversed bottom-to-top.
WCAG Failure : 2.4.3.
Expected result : The focus should follow the standard reading order of top-down, left-to-right.
Actual result : All of the elements in the chat window are focused in reverse order (bottom-up, left-to-right), making keyboard navigation very confusing. The same occurs with the reading order and scroll direction when the container has many messages (down moves up, up moves down).

Attachments : 

https://user-images.githubusercontent.com/66008453/140067653-52e33cd0-f04e-4495-ae80-111728ec0b1a.mp4

**Comment by nazam100** [comment]
Accessibility Issue : "+" (plus), 'Emoji', 'Send' Buttons has wrong button role, state not announced. Similar to the bug #5677. 
WCAG Failure : 4.1.2 
Expected result : The buttons are not a regular buttons. It functions as a '+'(plus), Emoji', 'Send' Buttons and should announce as such with a collapsed/expanded state. i.e. '+'Button, Emoji' button, 'Send' Button. 
Actual result :No '+' (plus). Emoji' button, 'Send' button role or  state is announced. Only a Clickable role are announced without any further context.

Attachments :


https://user-images.githubusercontent.com/66008453/140069788-a3e2066e-40d2-466f-b12c-6dd4c2f2086a.mp4

**Pull request merged**